### PR TITLE
Some query optimizations for contributor index view

### PR DESCRIPTION
**Pull request by karyon**
I looked at https://github.com/e-valuation/EvaP/issues/1509, but didn't find good opportunities for refactoring. I did some optimizations instead.

### CODE REVIEW

**Review by d-croft — APPROVED** [review]
Looks good to me.

File: evap/evaluation/models.py [hunk 1]
```diff
@@ -704,17 +704,15 @@ def hours_until_evaluation(self):
         return (self.vote_start_datetime - datetime.now()) / timedelta(hours=1)
 
     def is_user_editor_or_delegate(self, user):
-        represented_user_pks = [represented_user.pk for represented_user in user.represented_users.all()]
-        represented_user_pks.append(user.pk)
-        return self.contributions.filter(contributor__pk__in=represented_user_pks, role=Contribution.Role.EDITOR).exists() or self.course.responsibles.filter(pk__in=represented_user_pks).exists()
+        represented_users = user.represented_users.all() | UserProfile.objects.filter(pk=user.pk)
+        return self.contributions.filter(contributor__pk__in=represented_users, role=Contribution.Role.EDITOR).exists() or self.course.responsibles.filter(pk__in=represented_users).exists()
```
Comment (janno42) [inline evap/evaluation/models.py:708]: why is this `contributor__pk__in` and not `contributor__in`? you are now using the user profile objects, right?

**Review by janno42 — APPROVED** [review]
thank you! 👍

File: evap/evaluation/models.py (hunk 1 above)
Comment (karyon) [reply evap/evaluation/models.py:708]: Good question. Yes, I'm using the objects, and simply forgot to remove the __pk here. But it seems to work either way, the generated sql is identical. I assume this works to enable second case in this line, where the pk__ must be preserved...

**Pull request merged**